datalayer/main.go

```diff
@@ -0,0 +1,35 @@
+package datalayer
```

Comment:
Wrong place for this but I can't see how to comment on the name of a file. I think it's more typical to name a file after the package it provides. This is just from looking at some other Golang code I'm familiar with. So call this datalayer.go (and datalayer_test.go)